Mathlib/NumberTheory/NumberField/House.lean

```diff
@@ -0,0 +1,191 @@
+/-
+Copyright (c) 2024 Michail Karatarakis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michail Karatarakis
+-/
+
+import Mathlib.NumberTheory.NumberField.Embeddings
+
+/-!
+# House of an algebraic number
+This file defines the house of an algebraic number `α`, which is
+the largest modulus of its conjugates.
+
+## References
+* [D. Marcus, *Number Fields*][marcus1977number]
+* [Keng, H. L, *Introduction to number theory*][keng1982house]
+
+## Tags
+number field, algebraic number, house
+-/
+
+open BigOperators Module.Free Fintype NumberField Embeddings FiniteDimensional Matrix
+  Set Algebra Polynomial Basis Finset
+
+variable {K : Type*} [Field K] [NumberField K]
+
+local notation "h" => finrank ℚ K
+
+theorem Matrix.inv_mulVec_eq {u v : Fin h → ℂ} {M : Matrix (Fin h) (Fin h) ℂ} [Invertible M]
+    (hM : u = M.mulVec v) : M⁻¹.mulVec u = v := by
+  simp only [hM, Matrix.mulVec_mulVec, Matrix.inv_mul_of_invertible, Matrix.one_mulVec]
+
+theorem Finset.max'_eq_of_eq {γ : Type _} [LinearOrder γ] {s t : Finset γ} (hs : s.Nonempty)
+    (ht : t.Nonempty) (hst : s = t) : s.max' hs = t.max' ht := by simp_rw [hst]
+
+/-- `Finset.max'OfFintype` takes a function `f` from a finite and nonempty type `s` to a
+  linearly ordered type `γ`, and returns the maximum value of `f` over all elements of `s`. -/
+def Finset.max'OfFintype {s γ : Type _} [Fintype s] [Nonempty s] (f : Π _ : s, γ)
+    [LinearOrder γ] : γ := by
+  apply Finset.max' (f '' Set.univ).toFinset
+  simp only [Set.image_univ, Set.toFinset_range, Finset.image_nonempty]
+  exact Finset.univ_nonempty
+
+/-- `Matrix.max_abs_entry` takes a matrix `B` of size `finrank ℚ K` and returns the maximum
+  absolute value of its entries. This is done by first finding the maximum absolute value of the
+  entries in each row, and then finding the maximum of those row maxima. -/
+noncomputable def Matrix.maxAbsEntry (B : Matrix (Fin (finrank ℚ K)) (Fin (finrank ℚ K)) ℂ) := by
```

Comment:
The infinity norm of a matrix is already in mathlib, no need to redefine it. See for example [here](https://leanprover-community.github.io/mathlib4_docs/Mathlib/NumberTheory/SiegelsLemma.html) to see how to work with it.

Reply:
Great! I actually had a more elementary proof of Siegel's lemma for quite some time but then I saw the thread that people had PR'd a better version already.